### PR TITLE
bumps containerd to 1.6.18 and goland.org/x/net to 0.8.0 in provisioner

### PR DESCRIPTION
**Pull request by Disper**
**Description**

Aims to fix vulnerabilities reported by Whitesource in Provisioner

Changes proposed in this pull request:

- bumps containerd to 1.6.18
- bumps goland.org/x/net to 0.8.0

### CODE REVIEW

File: resources/kcp/values.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ global:
     tests:
       provisioner:
         dir:
-        version: "v20230426-ad965fac"
+          version: "PR-2718"
```
Comment (mvshao) [inline resources/kcp/values.yaml:20]: Is the indentation needed?

Comment (mvshao) [inline resources/kcp/values.yaml:20]: `PR-2718` doesn't exist in `europe-docker.pkg.dev/kyma-project/prod/control-plane`, see line 7. This image is in `/dev/control-plane/`

**Review by mvshao — CHANGES_REQUESTED** [review]

**Comment by Disper** [comment]
Added `/do-not-merge/hold` until the additional review will be done. @kyma-project/gopher could you see if you're fine with [the small structure change](https://github.com/kyma-project/control-plane/pull/2718/files#diff-61f7248e5a92221c2cd603ec6a9714363cfc901c42d661a71522c8910e6d3fef)?

**Comment by Disper** [comment]
I've received a hint from @szwedm that it might not work as we're trying to change override for this
https://github.com/kyma-project/control-plane/blob/main/resources/kcp/values.yaml#L7

See this:
https://github.com/kyma-project/control-plane/blob/main/resources/kcp/charts/provisioner/templates/deployment.yaml#L45

File: resources/kcp/charts/provisioner/values.yaml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 global:
+  containerRegistry:
+    path: europe-docker.pkg.dev/kyma-project
   images:
     provisioner:
-      dir:
-      version: "v20230412-91e5ada4"
+      version: "PR-17433"
```
Comment (koala7659) [inline resources/kcp/charts/provisioner/values.yaml:6]: Most likely copy-paste bug. Please change it to PR-2718

**Review by koala7659 — CHANGES_REQUESTED** [review]
Please see my comment and update image tag

**Review by koala7659 — APPROVED** [review]
LGTM

**Review by mvshao — APPROVED** [review]
LGTM

**Pull request merged**